### ISSUE TITLE
SharedMapper authorization

### ISSUE THREAD

**Issue by anton-shapovalov** [winsw/winsw#538]
### Feature Request

**Problem statement.**
We want to use WinSW for lunching Java applications as windows service and our applications requires SMB share, how can we configure windows auth to access SMB share. Now we can configure SMB share without auth in Active Directory but it isn't secure because all of machines and users on this domain can get the access to this share.

**Describe the solution you'd like.**

map enabled="false" label="N:" uncpath="\\UNC"/
map enabled="false" label="M:" uncpath="\\UNC2"/

Does WinSW have settings or options to specify windows user/password  in "mapping"?
Thanks

**Comment by oleg-nenashev** [comment]
No, mapping with credentials is not supported at the moment: https://github.com/winsw/winsw/blob/master/src/Plugins/SharedDirectoryMapper/SharedDirectoryMapperConfig.cs . Could be a good improvement

**Comment by NextTurn** [comment]
https://github.com/winsw/winsw/blob/dff63be343c0e7935c63577782ef8444a9e26495/src/Plugins/SharedDirectoryMapper/NativeMethods.cs#L12

I suggest stopping adding more XML attributes for credentials and introducing a referenceable element instead.

**Comment by MrGrymReaper** [comment]
@NextTurn What about the protection of the credentials stored in the configuration file for SharedMapper? Cause how you want them stored is unencrypted. Thus these details can be read by anyone or anything.

If you insist in storing the credentials in the config file, at least allow them to store them as encrypted, salted and/or peppered strings. Which the wrapper can read from the config file using a suitable program which can produce the appropriate strings.

**Comment by NextTurn** [comment]
> If you insist in storing the credentials in the config file

I don't want to store any credential in config files at all. My plan is to read credentials when installing and store them in the Windows Credential Manager.

**Comment by MrGrymReaper** [comment]
Alright sorry. However what are you doing about the other credential stored in the config file?

**Comment by NextTurn** [comment]
We'll just recommend people to stop storing credentials in config files...

**Comment by MrGrymReaper** [comment]
So you will just have them supply it a command line prompt then? If that is the case then the documentation would be helpful to be updated at that point.

**Comment by NextTurn** [comment]
FTR there was an idea to protect the whole config file (#439). It's not just about credentials. `<executable>` tampering is not less dangerous.

**Comment by MrGrymReaper** [comment]
That idea looks very good as that hardening will really help. Would be good to ensure Windows 10 (or Windows Server 2019) SCM support is present as it has a security hardening Group Policy option.

The idea is at least something similar to what happens with Systemd based daemons, having altered a daemon's definition file.

**Comment by NextTurn** [comment]
> So you will just have them supply it at a command line prompt then?

Not necessarily. I'll inform you when I finish the PR if you're interested.

**Comment by MrGrymReaper** [comment]
@NextTurn That would be great, as several pieces of software I use aren't designed as services but are servers. So being able to configure them using these features would be great. Also brought up this project's existence during Microsoft Build 2020, which I attended virtually.
